x/sync/network_server.go

```diff
@@ -136,32 +141,100 @@ func (s *NetworkServer) HandleChangeProofRequest(
 	}
 
 	// override limit if it is greater than maxKeyValuesLimit
-	limit := req.Limit
-	if limit > maxKeyValuesLimit {
-		limit = maxKeyValuesLimit
+	keyLimit := req.KeyLimit
+	if keyLimit > maxKeyValuesLimit {
+		keyLimit = maxKeyValuesLimit
 	}
+	bytesLimit := int(req.BytesLimit)
+	if bytesLimit > constants.DefaultMaxMessageSize {
+		bytesLimit = constants.DefaultMaxMessageSize
```

Comment:
This should be lower than the max message size because if the number of bytes is equal to (or just below) the max message size then this message will be too large

Reply:
I'm fine moving this to be a different constant.  What do you think an appropriate offset would be?

Reply:
The max size of the proof should be `[max avalanchego message size] - [app response overhead size]` but it's not entirely clear to me how to compute the latter given that the message is proto. To be safe maybe we could say  32 KiB? I doubt it's actually anywhere near that large but better safe than sorry, and I suspect that it won't matter in most cases since the proof won't be near the max size. We should probably put a TODO to return to this question